pkg/storage/stores/tsdb/index_client_test.go

```diff
@@ -214,3 +215,83 @@ func TestIndexClient_Stats(t *testing.T) {
 		})
 	}
 }
+
+func TestIndexClient_LabelVolume(t *testing.T) {
+	tempDir := t.TempDir()
+	tableRange := config.TableRange{
+		Start: 0,
+		End:   math.MaxInt64,
+		PeriodConfig: &config.PeriodConfig{
+			IndexTables: config.PeriodicTableConfig{
+				Period: config.ObjectStorageIndexRequiredPeriod,
+			},
+		},
+	}
+
+	indexStartToday := model.TimeFromUnixNano(time.Now().Truncate(config.ObjectStorageIndexRequiredPeriod).UnixNano())
+	indexStartYesterday := indexStartToday.Add(-config.ObjectStorageIndexRequiredPeriod)
+
+	tables := map[string][]*TSDBFile{
+		tableRange.PeriodConfig.IndexTables.TableFor(indexStartToday): {
+			BuildIndex(t, tempDir, []LoadableSeries{
+				{
+					Labels: mustParseLabels(`{foo="bar"}`),
+					Chunks: buildChunkMetas(int64(indexStartToday), int64(indexStartToday+99), 10),
+				},
+				{
+					Labels: mustParseLabels(`{fizz="buzz"}`),
+					Chunks: buildChunkMetas(int64(indexStartToday), int64(indexStartToday+99), 10),
+				},
+			}),
+		},
+
+		tableRange.PeriodConfig.IndexTables.TableFor(indexStartYesterday): {
+			BuildIndex(t, tempDir, []LoadableSeries{
+				{
+					Labels: mustParseLabels(`{foo="bar"}`),
+					Chunks: buildChunkMetas(int64(indexStartYesterday), int64(indexStartYesterday+99), 10),
+				},
+				{
+					Labels: mustParseLabels(`{foo="bar", fizz="buzz"}`),
+					Chunks: buildChunkMetas(int64(indexStartYesterday), int64(indexStartYesterday+99), 10),
+				},
+				{
+					Labels: mustParseLabels(`{ping="pong"}`),
+					Chunks: buildChunkMetas(int64(indexStartYesterday), int64(indexStartYesterday+99), 10),
+				},
+			}),
+		},
+	}
+
+	idx := newIndexShipperQuerier(mockIndexShipperIndexIterator{tables: tables}, config.TableRange{
+		Start:        0,
+		End:          math.MaxInt64,
+		PeriodConfig: &config.PeriodConfig{},
+	})
+
+	indexClient := NewIndexClient(idx, IndexClientOptions{UseBloomFilters: true})
+
+	t.Run("it returns label volumes from the whole index", func(t *testing.T) {
+		vol, err := indexClient.LabelVolume(context.Background(), "", indexStartYesterday, indexStartToday+1000, 10, nil...)
+		require.NoError(t, err)
+
+		require.Equal(t, &logproto.LabelVolumeResponse{
+			Volumes: []logproto.LabelVolume{
+				{Name: "foo", Value: "bar", Volume: 300 * 1024},
```

Comment:
nit: making these values `30 * 10240` make is a little more clear to the reader (IMO) where they come from, ie 3 of the streams added with a value of 10 each above must have matched. TBH I'm not sure where the `10240` comes from, I'm assuming it's in the test setup somewhere?

Reply:
[Each chunk is just 1 KB](https://github.com/grafana/loki/blob/4cebc2d85cc7b22526f4ba5f5364a2a9ee15aba2/pkg/storage/stores/tsdb/compactor_test.go#L197-L214) and each stream has 100 entries. 

`foo="bar"` is in three of the streams, `fizz="buzz"` is in two of them, etc